lib/reversetunnel/srv.go

```diff
@@ -260,13 +265,17 @@ func (cfg *Config) CheckAndSetDefaults() error {
 	if cfg.Component == "" {
 		cfg.Component = teleport.Component(teleport.ComponentProxy, teleport.ComponentServer)
 	}
-	logger := cfg.Log
 	if cfg.Log == nil {
-		logger = log.StandardLogger()
+		cfg.Log = log.StandardLogger()
 	}
-	cfg.Log = logger.WithFields(log.Fields{
-		teleport.ComponentKey: cfg.Component,
-	})
+	cfg.Log = cfg.Log.WithField(teleport.ComponentKey, cfg.Component)
+
+	if cfg.Logger == nil {
+		cfg.Logger = slog.Default()
+	}
+
+	cfg.Logger = cfg.Logger.With(teleport.ComponentKey, cfg.Component)
+
```

Comment:
nit: consistent spacing

```suggestion
	cfg.Log = cfg.Log.WithField(teleport.ComponentKey, cfg.Component)

	if cfg.Logger == nil {
		cfg.Logger = slog.Default()
	}
	cfg.Logger = cfg.Logger.With(teleport.ComponentKey, cfg.Component)

```

```suggestion

	cfg.Log = cfg.Log.WithField(teleport.ComponentKey, cfg.Component)

	if cfg.Logger == nil {
		cfg.Logger = slog.Default()
	}

	cfg.Logger = cfg.Logger.With(teleport.ComponentKey, cfg.Component)

```